test/Microsoft.Azure.WebJobs.Host.FunctionalTests/Queues/QueueListenerTests.cs

```diff
@@ -86,99 +83,65 @@ public QueueListenerTests(TestFixture fixture)
+                .Callback<ExceptionDispatchInfo>(e =>
                 {
-                    _original = value;
-                    _lastContentString = _original.AsString;
-                    _lastContentBytes = _original.AsBytes;
-                }
-            }
-
-            public override async Task<bool> BeginProcessingMessageAsync(CloudQueueMessage message, CancellationToken cancellationToken)
-            {
-                ValidateMessage(_original, message);
-                UpdateMessage(message);
-
-                return await base.BeginProcessingMessageAsync(message, cancellationToken);
-            }
-
-            public override async Task CompleteProcessingMessageAsync(CloudQueueMessage message, FunctionResult result, CancellationToken cancellationToken)
-            {
-                ValidateMessage(_original, message);
-                UpdateMessage(message);
-
-                await base.CompleteProcessingMessageAsync(message, result, cancellationToken);
-            }
-
-            protected override async Task DeleteMessageAsync(CloudQueueMessage message, CancellationToken cancellationToken)
-            {
-                ValidateMessage(_original, message);
-                UpdateMessage(message);
+                    throw e.SourceException;
+                });
 
-                await base.DeleteMessageAsync(message, cancellationToken);
-            }
+            QueueProcessorFactoryContext context = new QueueProcessorFactoryContext(queue, trace, null, queuesConfig);
+            Mock<QueueProcessor> mockProcessor = new Mock<QueueProcessor>(MockBehavior.Strict, context);
+            mockProcessor.CallBase = false;
 
-            protected override async Task CopyMessageToPoisonQueueAsync(CloudQueueMessage message, CloudQueue poisonQueue, CancellationToken cancellationToken)
-            {
-                ValidateMessage(_original, message);
-                UpdateMessage(message);
+            string messageContent = Guid.NewGuid().ToString();
+            CloudQueueMessage message = new CloudQueueMessage(messageContent);
+            await queue.AddMessageAsync(message, CancellationToken.None);
+            CloudQueueMessage messageFromCloud = await queue.GetMessageAsync();
 
-                await base.CopyMessageToPoisonQueueAsync(message, poisonQueue, cancellationToken);
-            }
+            QueueListener listener = new QueueListener(storageQueue, null, mockTriggerExecutor.Object, mockExceptionDispatcher.Object, trace,
+                null, null, queuesConfig);
 
-            protected override async Task ReleaseMessageAsync(CloudQueueMessage message, FunctionResult result, TimeSpan visibilityTimeout, CancellationToken cancellationToken)
-            {
-                ValidateMessage(_original, message);
-                UpdateMessage(message);
+            // Set up a function that waits 8 seconds (to allow renewal)
```

Comment:
Hmmm, ideally we wouldn't have a test that sleeps for 8 seconds. Maybe we don't even need this test. It seems below like you're mocking too much out anyways with the visibility renewal timer.